packages/kbn-observable/src/errors/empty_error.ts

```diff
@@ -0,0 +1,14 @@
+export class EmptyError extends Error {
+  code = 'KBN$_EMPTY_ERROR';
+
+  constructor(producer: string) {
+    super(
+      `EmptyError: ${producer} requires source stream to emit at least one value.`
+    );
+
+    // We're forching this to `any` as `captureStackTrace` is not a standard
+    // property, but a v8 specific one. There are node typings that we might
+    // want to use, see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/index.d.ts#L47
+    (Error as any).captureStackTrace(this, EmptyError);
```

Comment:
Does this mean we couldn't use the observable implementation in the browser? It seems like we should probably have some sort of fallback here.

Reply:
Ah, good point — this needs some extra work. I'll consider extracting something like `KbnError` from https://github.com/elastic/kibana/blob/5cddc100778f0dfd5c2e1f1342ab40a1cc7f52b9/src/ui/public/errors.js#L10 into some shared location, so we don't need to create a new top-level error class all the time.